src/core/database/query.helpers.ts

```diff
@@ -621,6 +621,22 @@ export function filterByUser(
   ]);
 }
 
+// used to search a specific engagement's relationship to the target base node
+// for example, searching all products a engagement is a part of
+export function filterByEngagement(
```

Comment:
This is specific to engagements. It should live in the engagements folder not core.